openqabot/types/incidents.py

```diff
@@ -72,11 +72,21 @@ def _is_scheduled_job(
                 ):
                     return True
             except KeyError:
-                log.debug(
-                    "Incident %s does not have %s arch in SLE-12 module version"
-                    % (inc.id, arch)
-                )
-                continue
+                try:
+                    if ver.startswith("12"):
+                        if (
+                            job["flavor"] == flavor
+                            and job["arch"] == arch
+                            and job["settings"]["REPOHASH"]
+                            == inc.revisions[ArchVer(arch, "12")]
+                        ):
+                            return True
+                except KeyError:
+                    log.debug(
+                        "Incident %s does not have %s arch in SLE-12 module version"
```

Comment:
At least a single sentence in the description explaining what is being addressed would be helpful here.

Also please add a unit test to check this case, since we had two commits namely #140 and #138 that did not fix it.

Reply:
The error handling is actually wrong. We have two places where we're checking for an unversioned module and where we check that this is 12. I'm looking to see if we can deduplicate this first.

Reply:
See #142